hubbard/hamiltonian.py

```diff
@@ -62,13 +61,18 @@ def __init__(self, TBHam, n=0, U=0.0, q=(0., 0.), nkpt=[1, 1, 1], kT=1e-5):
             ntot = len(self.geometry)
 
         self.q = np.array(q, dtype=np.float64).copy()
-        assert len(self.q) == 2 # Users *must* specify two values
+        # Ensure the length of q is equal to the dimension of the problem
+        self.q = self.q[:self.H.dim]
```

Comment:
`self.q = self.q[:self.H.spin.spins]`?

Reply:
Actually it needs to be `min(2, self.H.spin.spins)`.

I will add a new flag to `Spin` to get the number of spinors

Then it would be `self.q = self.q[:self.H.spin.spinor]`

Reply:
Now this is handled by saving the spin degree of freedom in the `HubbardHamiltonian` object, and by setting a maximum of 2 since we have only implemented the cases for either `polarized` or `unpolarized` Hamiltonians.
See commit a5632f0.